cpp/cmake_modules/SetupCxxFlags.cmake

```diff
@@ -48,9 +48,89 @@ if (MSVC)
     set(CXX_COMMON_FLAGS "/W3")
   endif()
 else()
-  set(CXX_COMMON_FLAGS "-Wall -std=c++11")
+  # Common flags set below with warning level
+  set(CXX_COMMON_FLAGS "")
 endif()
 
+# Build warning level (CHECKIN, EVERYTHING, etc.)
+
+# GCC/Clang warning flags by version:
+# https://github.com/Barro/compiler-warnings
```

Comment:
What is the reference for? This repo has a GPL license in it, so want to make sure no IP from that codebase is being taken in here

Reply:
I just included as a way for people to see which build of Clang introduced a warning; I'll remove it. I came up with the list that I use from compiling with -Weverything and suppressing warnings until the existing code base built with the version of the Clang compiler that I have.

Reply:
CHECKIN adds "-Werror" and a few other warnings like the Travis CI build.

Currently, warnings are not treated as errors when building on your own development machine. Then, you check in your code and Travis CI etc. add "-Werror" and a few more warnings.

I'll change "CHECKIN" to "PRODUCTION".

"DEVELOPMENT" does not treat warnings as errors.

Reply:
Sorry, I miscommunicated. We should not fail users' builds because of compiler warnings. But developers, yes. So

Default: PRODUCTION (warnings to not cause errors)
Development: CHECKIN (warnings -> errors)
Pedantic: EVERYTHING

You can leave the reference to that repo as long as it is clear that no code is being redistributed